### PR TITLE
Make the upgrade code tolerant of pre-release versions

### PR DESCRIPTION
**Pull request by WPFerg**

### CODE REVIEW

File: src/store-service.js [hunk 1]
```diff
@@ -130,8 +130,8 @@
             var thisVersion = parseVersion(this.version);
             var storedVersion = parseVersion(localStorage.getItem(VERSION_KEY));
 
-            // If there's more than x.y.z, someone's probably tampered.
-            if (storedVersion.length > 3) {
+            // If there's more than x.y.z-rc.a, someone's probably tampered.
+            if (storedVersion.length > 4) {
```
Comment (bjedrzejewski) [inline src/store-service.js]: Does this check gives us anything other than potential problems?

Comment (bjedrzejewski) [inline src/store-service.js]: Isn't this - return thisVersion[0] !== storedVersion[0]; enough? :)

Comment (WPFerg) [inline src/store-service.js]: Probably not, aside from being able to check a malformed version number.

Comment (bjedrzejewski) [inline src/store-service.js]: I think we should remove it, this is beyond the scope of the method 'should upgrade'. Unless there is a good reason to keep it.

Comment (WPFerg) [inline src/store-service.js]: I've removed it.

**Pull request merged**